test/e2e/index.ts

```diff
@@ -0,0 +1,145 @@
+
+const { BlindRSAMode, Client, verifyToken } = publicVerif;
+
+interface MTLSConfiguration {
+	certPath: string;
+	keyPath: string;
+}
+
+export function base64UrlToUint8Array(base64Url: string): Uint8Array {
+	// Convert URL escaped characters to regular base64 string
+	const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+
+	return Uint8Array.from(atob(base64), c => c.charCodeAt(0));
+}
+
+async function fetchWithMTLS(mTLS: MTLSConfiguration): Promise<typeof fetch> {
+	const clientCert = fs.readFileSync(mTLS.certPath);
+	const clientKey = fs.readFileSync(mTLS.keyPath);
+
+	const agent = new https.Agent({
+		cert: clientCert,
+		key: clientKey,
+		rejectUnauthorized: false,
+	});
+
+	return (request, init) => fetch(request, { ...init, agent });
+}
+
+async function getIssuerConfig(name: string, mTLS?: MTLSConfiguration) {
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(`https://${name}${PRIVATE_TOKEN_ISSUER_DIRECTORY}`);
+	const config: IssuerConfig = (await response.json()) as IssuerConfig;
+
+	const token = config['token-keys'].find(
+		token => token['token-type'] == TOKEN_TYPES.BLIND_RSA.value
+	);
+
+	if (!token) {
+		throw new Error('Could not find BlindRSA token key on issuer');
+	}
+
+	const publicKeyEnc = base64UrlToUint8Array(token['token-key']);
+	const publicKey = await importPublicKey(util.convertRSASSAPSSToEnc(publicKeyEnc));
+
+	return {
+		publicKey,
+		publicKeyEnc,
+		url: `https://${name}${config['issuer-request-uri']}`,
+	};
+}
+
+async function importPublicKey(spki: Uint8Array) {
+	return crypto.subtle.importKey('spki', spki, { name: 'RSA-PSS', hash: 'SHA-384' }, true, [
+		'verify',
+	]);
+}
+
+async function test(issuerName: string, mTLS?: MTLSConfiguration) {
+	console.log(`
+████████╗███████╗███████╗████████╗   ███████╗██████╗ ███████╗
+╚══██╔══╝██╔════╝██╔════╝╚══██╔══╝██╗██╔════╝╚════██╗██╔════╝
+   ██║   █████╗  ███████╗   ██║   ╚═╝█████╗   █████╔╝█████╗  
+   ██║   ██╔══╝  ╚════██║   ██║   ██╗██╔══╝  ██╔═══╝ ██╔══╝  
+   ██║   ███████╗███████║   ██║   ╚═╝███████╗███████╗███████╗
+   ╚═╝   ╚══════╝╚══════╝   ╚═╝      ╚══════╝╚══════╝╚══════╝
+`);
+	console.log('Testing with issuer', issuerName);
+	const client = new Client(BlindRSAMode.PSS);
+
+	const redemptionContext = new Uint8Array(32);
+	redemptionContext.fill(0xfe);
+	const challenge = new TokenChallenge(TOKEN_TYPES.BLIND_RSA.value, issuerName, redemptionContext);
+
+	const {
+		publicKeyEnc: issuerPublicKeyEnc,
+		publicKey: issuerPublicKey,
+		url: issuerRequestURL,
+	} = await getIssuerConfig(issuerName);
+
+	const tokenRequest = await client.createTokenRequest(challenge, issuerPublicKeyEnc);
+
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(issuerRequestURL, {
+		method: 'POST',
+		headers: {
+			'Content-Type': MediaType.PRIVATE_TOKEN_REQUEST,
+			'Accept': MediaType.PRIVATE_TOKEN_RESPONSE,
+		},
+		body: Buffer.from(tokenRequest.serialize()),
```

Comment:
Can we remove Buffer?
```suggestion
		body: tokenRequest.serialize().buffer,
```

Reply:
I've updated the type to use .buffer and cast `Buffer`. This is a [type error](https://github.com/node-fetch/node-fetch/pull/1521) on `node-fetch` side.